### PR TITLE
Possibility to duplicate scenarios Issue Solved

### PR DESCRIPTION
**Pull request by AnujSaha0111**
When creating a new scenario by duplicating an existing one, the code checks whether any Data Nodes (inputs) for the tasks have changed. If no changes are detected, it reuses the cached task output instead of re-running the task, improving performance. Finally, the new scenario is saved with the appropriate reused or re-executed tasks.

### CODE REVIEW

File: Data_Node_Changes_check.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class ScenarioManager:
```
Comment (trgiangdo) [inline Data_Node_Changes_check.py:1]: We already have a `_ScenarioManager` class

File: Scenario_Duplication_Method.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    def duplicate_scenario(self, original_scenario_id):
+        # Fetch the original scenario by ID
+        original_scenario = self.get_scenario_by_id(original_scenario_id)
+
+        # Create a copy of the scenario structure
+        new_scenario = Scenario()
+        new_scenario.name = f"Duplicate of {original_scenario.name}"
+        
+        # Iterate over tasks in the original scenario
```
Comment (trgiangdo) [inline Scenario_Duplication_Method.py:11]: There are a lot more properties in a Scenario than just `tasks`

We also need to duplicate those

File: Reusing_Cached_Task_Results.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Scenario:
+
+    def add_task_with_cached_output(self, task):
+        # Instead of executing the task, add it with cached results
+        cached_result = self.get_cached_task_output(task.id)
+        self.tasks.append({
+            'task': task,
+            'output': cached_result
+        })
```
Comment (trgiangdo) [inline Reusing_Cached_Task_Results.py]: This doesn't make any sense regarding the Taipy `Task` object. A scenario `tasks` is not just a list of dictionary with output

**Review by trgiangdo — CHANGES_REQUESTED** [review]
Thank you for interested in contributing to Taipy.

Duplicating a scenario is a complicated topic, which requires duplicating every properties of a Scenario object, including children objects as well. Please take a look at the `_ScenarioManager` and `Scenario` classes in Taipy for more detail.

There are a lot of placeholder methods in your PR, and it's not working yet, so I believe this is not ready to be merged

File: Data_Node_Changes_check.py (hunk 1 above)
Comment (AnujSaha0111) [reply Data_Node_Changes_check.py:1]: For this file, I needed a classs, so I added

File: Scenario_Duplication_Method.py (hunk 2 above)
Comment (AnujSaha0111) [reply Scenario_Duplication_Method.py:11]: I found this helpful

File: Reusing_Cached_Task_Results.py (hunk 3 above)
Comment (AnujSaha0111) [reply Reusing_Cached_Task_Results.py]: deleted it

**Pull request closed without merging**